clients/src/main/java/org/apache/kafka/clients/KafkaClient.java

```diff
@@ -86,7 +86,17 @@
     List<ClientResponse> poll(long timeout, long now);
 
     /**
+     * Diconnects the connection to a particular node, if there is one.
+     * Any pending ClientRequests for this connection will receive disconnections.
+     *
+     * @param nodeId The id of the node
+     */
+    void disconnect(String nodeId);
```

Comment:
Also, as @ijuma said, when we implement per-request timeouts in NetworkClient, AdminClient won't need to call disconnect any more.